In [1]:
import pandas as pd
import requests

# Collecting Job Data Using APIs - Issue with API call on Key Skill C 

This notebook has been created to illustrate the issue I found to retrieve jobs data calling the Jobs API. 

> To run the Notebook, you need to run the associate `Jobs_API.ipynb` and make sure that the `job.json` file is in the same folder of the Notebook. The `job.json` file has been downloaded from https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBM-DA0321EN-SkillsNetwork/labs/module%201/Accessing%20Data%20Using%20APIs/jobs.json link

## Number of job data based on Technology

The following function returns the number of jobs matching the desired Technology

In [2]:
api_url="http://127.0.0.1:5000/data"
def get_number_of_jobs_T(technology):
    payload={"Key Skills": technology}
    response=requests.get(api_url, params=payload)
    if response.ok:
        number_of_jobs = len(response.json())
    #your code goes here
    return technology,number_of_jobs

- Retrieving the number of jobs for `Python` technology

In [3]:
get_number_of_jobs_T("Python")

('Python', 1173)

Now, if we dig deep in the function, `get_number_of_jobs_T`, we can test if the result is consistent with the desired behavior

In [4]:
tech_test = 'Python'
payload_test={"Key Skills": tech_test}
response_test = requests.get(api_url, params=payload_test)

Let's see what `response_test` looks like examinating the json dump

In [7]:
response_test.json()[:3]

[{'Functional Area': 'IT Hardware , Technical Support , Telecom Engineering',
  'Id': 38,
  'Industry': 'IT-Software, Software Services',
  'Job Experience Required': '15 - 20 yrs',
  'Job Title': 'nutanix sme',
  'Key Skills': 'Troubleshooting| DNS| DHCP| Monitoring| Virtualization| Shell scripting|Python| VMS| Open source| Archiving',
  'Location': 'Washington DC',
  'Role': 'Technical Support Manager',
  'Role Category': 'Technical Support'},
 {'Functional Area': 'IT Software - Application Programming , Maintenance',
  'Id': 73,
  'Industry': 'IT-Software, Software Services',
  'Job Experience Required': '6 - 11 yrs',
  'Job Title': 'Openstack Engineer / Openstack Admin with Ceph Storage',
  'Key Skills': 'cloud| puppet|Python| git| KVM| linux| jenkins| openstack| ansible',
  'Location': 'Washington DC',
  'Role': 'System Analyst',
  'Role Category': 'Programming & Design'},
 {'Functional Area': 'Analytics & Business Intelligence',
  'Id': 117,
  'Industry': 'IT-Software, Software S

More specifically, if we have a look to the `Key Skills` key we will find:

In [9]:
for i in range(40,50):
    print(response_test.json()[i]['Key Skills'])

C++| Linux| Information security| Debugging| Application development| Windows| Troubleshooting| Virtualization| SQL|Python
Unix| Linux| MySQL| Computer science| Software design| DNS| Data structures| Perl| HTTP|Python
SQL| Javascript|Python| rest| development| gaming| technical| software| testing| mobile| it| CSS3| Technical documentation| quality| Backend| Front end| Web services| HP data protector| ui| web| design| Software development life cycle| developer
Chef| Unix| Linux| TCP| Automation|Python| CMS| Troubleshooting| DNS| Ruby
Javascript| Application development|Python| Maven| development| css| DOM| software| JSON| nosql| spring| database| Version control| GIT| java| automation| RDBMS| Analytical| developer| api| mongodb
Selenium| Android| Java| Automation Testing| Jenkins| Continuous Integration|Python| GIT| SVN| Agile Development
Linux| HTML| SQL| Computer science| Coding| MySQL| Windows| Analytics| Android|Python
Python| Business intelligence| Coding| Computer science| Machine

So, Python is a Skill we were looking for.

Now let's try with `C` as Technology

In [11]:
get_number_of_jobs_T("C")

('C', 13498)

So, it looks like we have 13498 jobs data. But if we check further, as in the previous case of using `Python` as Technology, we will find:

In [12]:
tech_test = 'C'
payload_test={"Key Skills": tech_test}
response_test = requests.get(api_url, params=payload_test)

In [13]:
print(len(response_test.json()))

13498


In [14]:
for i in range(40,50):
    print(response_test.json()[i]['Key Skills'])

Revenue Cycle Management| US Healthcare| Payment Posting| Medical Billing| Medical Coding| us visa| Team Leading Skills| Healthcare| Denial Management| RCM
accounts executive| accountant| Stock Management| gst| auditing| Statutory Compliance| accounting| Account Assistant| Bank Reconciliation| tax audit| banking| monthly reports| Payroll| salary processing| income tax| Income Tax Return| taxation
Informatica Power Center| Informatica
Validation| Installation| Breakdown Maintenance| Industrial Safety| gmp| Boiler| Design| HVAC| Gensets| Air Compressor| Breakdown| RO| who
Pay Per Click| Google AdWords| Search Engine Marketing| PPC| SEM| Google Analytics| Website Analysis| Yahoo| Web Analytics| Marketing Initiatives
Project management| PPAP| Automotive| External audit| Conflict resolution| Resource mobilization| PFMEA| Risk management| Project life cycle
PR| Public Relations| Media Relations| Social Media| Marketing Communication| corporate communication| media relation| corp comm| Online

In this case, we don't find `C` in the `Key Skills` key. What am I doing wrong?